<a href="https://colab.research.google.com/github/Samikshya-mohanty/datascience_assignments/blob/main/Samikshya_Mohanty__CPDA_PGDSE_Batch1_ipnyb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
import numpy as np
import math as mth
from scipy import stats
import statsmodels.api as sm
from statsmodels.formula.api import ols


In [3]:
df = pd.read_csv('/content/property.csv')

In [4]:
df.head()

,Suburb,Address,Rooms,Type,Price,Method,SellerG,Date,Distance,Postcode,...,Bathroom,Car,Landsize,BuildingArea,YearBuilt,CouncilArea,Lattitude,Longtitude,Regionname,Propertycount
0,Abbotsford,85 Turner St,2,h,1480000.0,S,Biggin,3/12/2016,2.5,3067.0,...,1.0,1.0,202.0,NaN,NaN,Yarra,-37.7996,144.9984,Northern Metropolitan,4019.0
1,Abbotsford,25 Bloomburg St,2,h,1035000.0,S,Biggin,4/02/2016,2.5,3067.0,...,1.0,0.0,156.0,79.0,1900.0,Yarra,-37.8079,144.9934,Northern Metropolitan,4019.0
2,Abbotsford,5 Charles St,3,h,1465000.0,SP,Biggin,4/03/2017,2.5,3067.0,...,2.0,0.0,134.0,150.0,1900.0,Yarra,-37.8093,144.9944,Northern Metropolitan,4019.0
3,Abbotsford,40 Federation La,3,h,850000.0,PI,Biggin,4/03/2017,2.5,3067.0,...,2.0,1.0,94.0,NaN,NaN,Yarra,-37.7969,144.9969,Northern Metropolitan,4019.0
4,Abbotsford,55a Park St,4,h,1600000.0,VB,Nelson,4/06/2016,2.5,3067.0,...,1.0,2.0,120.0,142.0,2014.0,Yarra,-37.8072,144.9941,Northern Metropolitan,4019.0


In [5]:
df.shape

(13580, 21)

Q1. For the suburb of Altona, it is postulated that a typical property sells for $800,000.


Use the data at hand to test this assumption.
Is the typical property price really $800,000 or has it increased?
Use a significance level of 5%.

Null hypothesis: mean <= $800,000

Alternate Hypothesis: mean > $800,000

For this we will do one sample right tailed T - test.

In [6]:
df_altona = df[df['Suburb'] == 'Altona']

In [7]:
df_altona

,Suburb,Address,Rooms,Type,Price,Method,SellerG,Date,Distance,Postcode,...,Bathroom,Car,Landsize,BuildingArea,YearBuilt,CouncilArea,Lattitude,Longtitude,Regionname,Propertycount
141,Altona,158 Queen St,3,h,520000.0,VB,Greg,3/09/2016,13.8,3018.0,...,2.0,1.0,352.0,242.0,2015.0,Hobsons Bay,-37.87000,144.82500,Western Metropolitan,5301.0
142,Altona,59 Bracken Gr,5,h,1525000.0,S,Greg,3/12/2016,13.8,3018.0,...,3.0,3.0,729.0,268.0,2005.0,Hobsons Bay,-37.87210,144.81050,Western Metropolitan,5301.0
143,Altona,1/123 Blyth St,2,t,720000.0,S,hockingstuart,4/02/2016,13.8,3018.0,...,1.0,2.0,292.0,125.0,2013.0,Hobsons Bay,-37.86870,144.81970,Western Metropolitan,5301.0
144,Altona,4 Blyth St,3,h,1120000.0,S,Barlow,4/03/2017,13.8,3018.0,...,1.0,2.0,506.0,120.0,1930.0,Hobsons Bay,-37.86930,144.83750,Western Metropolitan,5301.0
145,Altona,20 Linnet St,4,h,780000.0,PI,Sweeney,4/06/2016,13.8,3018.0,...,1.0,4.0,655.0,NaN,NaN,Hobsons Bay,-37.86800,144.81540,Western Metropolitan,5301.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
12216,Altona,86 Grieve Pde,3,h,913000.0,S,Greg,3/09/2017,11.0,3018.0,...,2.0,3.0,530.0,148.0,1985.0,NaN,-37.86140,144.81931,Western Metropolitan,5301.0
12432,Altona,7/35 Grieve Pde,2,u,458000.0,SP,hockingstuart,9/09/2017,11.0,3018.0,...,1.0,1.0,63.0,75.0,1975.0,NaN,-37.86933,144.81704,Western Metropolitan,5301.0
12634,Altona,6 Myrtle Gr,3,h,1248500.0,S,RT,16/09/2017,11.0,3018.0,...,1.0,3.0,710.0,NaN,NaN,NaN,-37.87182,144.81600,Western Metropolitan,5301.0
13338,Altona,20 Burt St,3,h,920000.0,VB,PRD,26/08/2017,11.0,3018.0,...,2.0,2.0,584.0,NaN,NaN,NaN,-37.86249,144.82162,Western Metropolitan,5301.0


We have filtered the data for the suburb of Altona and got 74 samples.

In [8]:
df_altona['Price'].describe()

,Price
count,7.400000e+01
mean,8.348304e+05
std,2.915460e+05
min,3.910000e+05
25%,6.225000e+05
50%,7.735000e+05
75%,9.725000e+05
max,1.780000e+06


T test formula

$$t = \frac{\bar{x} - \mu_0}{s / \sqrt{n}}$$

In [9]:
sample_mean = df_altona['Price'].mean()
sample_std_deviation = df_altona['Price'].std()
sample_size = len(df_altona)
null_hypothesis = 800000


In [10]:
t_value = (sample_mean - null_hypothesis) / (sample_std_deviation / mth.sqrt(sample_size))
t_value

np.float64(1.0277020770199679)

In [11]:
t_stat, p_value = stats.ttest_1samp(df_altona['Price'], null_hypothesis, alternative='greater')

In [12]:
t_stat

np.float64(1.0277020770199676)

In [13]:
p_value

np.float64(0.1537416356527775)

As the P-value 0.15 is greater than significant value 0.05, hence we fail to  reject the null hypothesis.

There is not enough evidence to claim that the price has increased beyond $\$800,000$.

Q2. For the year 2016, is there any difference in the prices of properties sold in the
summer months vs winter months?

• Consider months from October till March as winter months and rest as
summer months.

• Use a significance level of 5%.


Null Hypothesis : Avg price of summer_mean = Avg price of winter_mean

Alternate Hypothesis :  Avg price of summer_mean != Avg price of winter_mean

Independent Two-Sample T-test

In [14]:
df.head()

,Suburb,Address,Rooms,Type,Price,Method,SellerG,Date,Distance,Postcode,...,Bathroom,Car,Landsize,BuildingArea,YearBuilt,CouncilArea,Lattitude,Longtitude,Regionname,Propertycount
0,Abbotsford,85 Turner St,2,h,1480000.0,S,Biggin,3/12/2016,2.5,3067.0,...,1.0,1.0,202.0,NaN,NaN,Yarra,-37.7996,144.9984,Northern Metropolitan,4019.0
1,Abbotsford,25 Bloomburg St,2,h,1035000.0,S,Biggin,4/02/2016,2.5,3067.0,...,1.0,0.0,156.0,79.0,1900.0,Yarra,-37.8079,144.9934,Northern Metropolitan,4019.0
2,Abbotsford,5 Charles St,3,h,1465000.0,SP,Biggin,4/03/2017,2.5,3067.0,...,2.0,0.0,134.0,150.0,1900.0,Yarra,-37.8093,144.9944,Northern Metropolitan,4019.0
3,Abbotsford,40 Federation La,3,h,850000.0,PI,Biggin,4/03/2017,2.5,3067.0,...,2.0,1.0,94.0,NaN,NaN,Yarra,-37.7969,144.9969,Northern Metropolitan,4019.0
4,Abbotsford,55a Park St,4,h,1600000.0,VB,Nelson,4/06/2016,2.5,3067.0,...,1.0,2.0,120.0,142.0,2014.0,Yarra,-37.8072,144.9941,Northern Metropolitan,4019.0


In [15]:
# Filtering the data for the year 2016
df['Date'] = pd.to_datetime(df['Date'], format='%d/%m/%Y')
df_2016 = df[df['Date'].dt.year == 2016]

In [16]:
df_2016

,Suburb,Address,Rooms,Type,Price,Method,SellerG,Date,Distance,Postcode,...,Bathroom,Car,Landsize,BuildingArea,YearBuilt,CouncilArea,Lattitude,Longtitude,Regionname,Propertycount
0,Abbotsford,85 Turner St,2,h,1480000.0,S,Biggin,2016-12-03,2.5,3067.0,...,1.0,1.0,202.0,NaN,NaN,Yarra,-37.7996,144.9984,Northern Metropolitan,4019.0
1,Abbotsford,25 Bloomburg St,2,h,1035000.0,S,Biggin,2016-02-04,2.5,3067.0,...,1.0,0.0,156.0,79.0,1900.0,Yarra,-37.8079,144.9934,Northern Metropolitan,4019.0
4,Abbotsford,55a Park St,4,h,1600000.0,VB,Nelson,2016-06-04,2.5,3067.0,...,1.0,2.0,120.0,142.0,2014.0,Yarra,-37.8072,144.9941,Northern Metropolitan,4019.0
5,Abbotsford,129 Charles St,2,h,941000.0,S,Jellis,2016-05-07,2.5,3067.0,...,1.0,0.0,181.0,NaN,NaN,Yarra,-37.8041,144.9953,Northern Metropolitan,4019.0
6,Abbotsford,124 Yarra St,3,h,1876000.0,S,Nelson,2016-05-07,2.5,3067.0,...,2.0,0.0,245.0,210.0,1910.0,Yarra,-37.8024,144.9993,Northern Metropolitan,4019.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
7426,Gardenvale,7/32 Gardenia Rd,2,u,440000.0,S,Biggin,2016-06-18,9.8,3185.0,...,1.0,1.0,0.0,75.0,NaN,Glen Eira,-37.8971,145.0093,Southern Metropolitan,534.0
7427,Gardenvale,1/18 Magnolia Rd,2,u,595000.0,S,Biggin,2016-11-19,9.8,3185.0,...,1.0,1.0,0.0,75.0,1960.0,Glen Eira,-37.8994,145.0101,Southern Metropolitan,534.0
7428,Gardenvale,25 Magnolia Rd,4,h,2200000.0,S,hockingstuart,2016-08-28,9.8,3185.0,...,3.0,3.0,474.0,230.0,2010.0,Glen Eira,-37.8988,145.0098,Southern Metropolitan,534.0
7429,Princes Hill,76 Arnold St,3,h,1700000.0,VB,Collins,2016-11-27,3.7,3054.0,...,1.0,2.0,341.0,NaN,NaN,Yarra,-37.7832,144.9655,Northern Metropolitan,1008.0


In [17]:
# separating winter and and summer months prices
winter_months = [10, 11, 12, 1, 2, 3]
winter_prices = df_2016[df_2016['Date'].dt.month.isin(winter_months)]['Price'].dropna()
summer_prices = df_2016[~df_2016['Date'].dt.month.isin(winter_months)]['Price'].dropna()

In [18]:
def get_metrics(df:pd.Series):
  sample_mean = df.mean()
  sample_std_deviation = df.std()
  sample_size = len(df)
  return sample_mean, sample_std_deviation, sample_size


In [19]:
summer_mean, summer_std_deviation, summer_sample_size = get_metrics(summer_prices)
summer_mean, summer_std_deviation, summer_sample_size

(np.float64(1048054.7286917741), 621493.4398118115, 4036)

In [20]:
winter_mean, winter_std_deviation, winter_sample_size = get_metrics(winter_prices)
winter_mean, winter_std_deviation, winter_sample_size

(np.float64(1116647.5917391304), 695498.2768521258, 2300)

$$t = \frac{\bar{x}_1 - \bar{x}_2}{\sqrt{\frac{s_1^2}{n_1} + \frac{s_2^2}{n_2}}}$$

In [21]:
t_value = (summer_mean - winter_mean) / mth.sqrt((summer_std_deviation**2 / summer_sample_size) + (winter_std_deviation**2 / winter_sample_size))
t_value

np.float64(-3.9211012778003775)

In [22]:
t_stat, p_value = stats.ttest_ind(winter_prices, summer_prices, equal_var=False, alternative='two-sided')

In [23]:
t_stat, p_value

(np.float64(3.9211012778003753), np.float64(8.950129747463378e-05))

Since the p-value ($0.0001$) is significantly lower than the threshold of $0.05$, we reject the null hypothesis.

And, we conclude that, there is a significant difference in the prices of properties sold in the summer months vs the winter months in 2016

Q3. For the suburb of Abbotsford, what is the probability that out of 10 properties sold,
3 will not have a car parking space?

• Use the column car in the dataset.

• Round off your answer to 3 decimal places.

In [24]:
df_Abbotsford = df[df['Suburb'] == 'Abbotsford']

In [25]:
df_Abbotsford.head()

,Suburb,Address,Rooms,Type,Price,Method,SellerG,Date,Distance,Postcode,...,Bathroom,Car,Landsize,BuildingArea,YearBuilt,CouncilArea,Lattitude,Longtitude,Regionname,Propertycount
0,Abbotsford,85 Turner St,2,h,1480000.0,S,Biggin,2016-12-03,2.5,3067.0,...,1.0,1.0,202.0,NaN,NaN,Yarra,-37.7996,144.9984,Northern Metropolitan,4019.0
1,Abbotsford,25 Bloomburg St,2,h,1035000.0,S,Biggin,2016-02-04,2.5,3067.0,...,1.0,0.0,156.0,79.0,1900.0,Yarra,-37.8079,144.9934,Northern Metropolitan,4019.0
2,Abbotsford,5 Charles St,3,h,1465000.0,SP,Biggin,2017-03-04,2.5,3067.0,...,2.0,0.0,134.0,150.0,1900.0,Yarra,-37.8093,144.9944,Northern Metropolitan,4019.0
3,Abbotsford,40 Federation La,3,h,850000.0,PI,Biggin,2017-03-04,2.5,3067.0,...,2.0,1.0,94.0,NaN,NaN,Yarra,-37.7969,144.9969,Northern Metropolitan,4019.0
4,Abbotsford,55a Park St,4,h,1600000.0,VB,Nelson,2016-06-04,2.5,3067.0,...,1.0,2.0,120.0,142.0,2014.0,Yarra,-37.8072,144.9941,Northern Metropolitan,4019.0


In [26]:
df_Abbotsford['Car'].value_counts()

,count
Car,
1.0,29
0.0,15
2.0,12


As per the above data, probability that property has a car parking = property having car space / total_no_property

In [27]:
prob_car_space = (df_Abbotsford['Car'].value_counts()[1]+df_Abbotsford['Car'].value_counts()[2]) / len(df_Abbotsford)
prob_car_space

np.float64(0.7321428571428571)

In [28]:
prob_nocar_space = 1 - prob_car_space
prob_nocar_space

np.float64(0.2678571428571429)

This follows a Binomial Distribution. First, we find the probability ($p$) of a single property having no parking ($Car = 0$) in Abbotsford.

P(X = r) = nCr pr (1-p) n-r, r = 0, 1, 2, 3....



In [29]:
round(mth.comb(10, 3) * (prob_nocar_space**3) * (prob_car_space**7),3)


np.float64(0.26)

The probability that out of 10 properties sold,
3 will not have a car parking space is 0.26

Q4.  In the suburb of Abbotsford, what are the chances of finding a property with 3
rooms? Round your answer to 3 decimal places.

In [30]:
df_Abbotsford['Rooms'].value_counts()

,count
Rooms,
2,27
3,20
1,5
4,4


In [31]:
probabilty_3rooms = round(df_Abbotsford['Rooms'].value_counts()[3] / len(df_Abbotsford),3)
probabilty_3rooms

np.float64(0.357)

The chances of finding a property with 3 rooms is 0.357

Q5. In the suburb of Abbotsford, what are the chances of finding a property with 2
bathrooms? Round your answer to 3 decimal places.

In [32]:
df_Abbotsford['Bathroom'].value_counts()

,count
Bathroom,
1.0,35
2.0,19
3.0,2


In [33]:
Probability_2bathrooms = round(df_Abbotsford['Bathroom'].value_counts()[2] / len(df_Abbotsford),3)
Probability_2bathrooms

np.float64(0.339)

The chances of finding a property with 2 bathrooms is 0.339

Q6. One-Sample Hypothesis Test (Industry Pricing) A real estate firm claims that the
average property price in Richmond is $1,000,000. Using the dataset, test whether
the actual average price is significantly different from this claim at a 5% significance level. Clearly state:

• Null and alternative hypotheses

• Test statistic

• p-value

• Final business conclusion

Null hypothesis = $1,000,000

Alternate hypothesis != $1,000,000

Significance = 0.05

In [34]:
df_Richmond = df[df['Suburb'] == 'Richmond']

In [35]:
df_Richmond['Price'].describe()

,Price
count,2.600000e+02
mean,1.083564e+06
std,5.223535e+05
min,2.700000e+05
25%,6.840000e+05
50%,1.078000e+06
75%,1.350000e+06
max,3.335000e+06


$$t = \frac{\bar{x} - \mu_0}{\frac{s}{\sqrt{n}}}$$

In [36]:
sample_mean = df_Richmond['Price'].mean()
sample_std_deviation = df_Richmond['Price'].std()
sample_size = len(df_Richmond)
null_hypothesis = 1000000

In [37]:
sample_mean, sample_std_deviation, sample_size

(np.float64(1083564.423076923), 522353.52451109583, 260)

In [38]:
t_value_Richmond = (sample_mean - null_hypothesis) / (sample_std_deviation / mth.sqrt(sample_size))
t_value_Richmond

np.float64(2.5795477040749226)

In [39]:
t_stat, p_value = stats.ttest_1samp(df_Richmond['Price'], null_hypothesis, alternative='two-sided')

In [40]:
t_stat,p_value

(np.float64(2.579547704074923), np.float64(0.01044499066415202))

Since the p-value 0.010 is less than the significance value 0.05, we reject the null hypothesis.

This suggest the avg price of property sold is not equal to 1000000

Q7. Independent Two-Sample T-Test (Feature Impact)
Do properties with car parking sell at a higher average price than properties without
car parking, across the entire dataset? Use a 5% significance level and justify:

• Choice of test

• Interpretation of p-value

• Business implications for developers

Null hypthesis: avg_price_car_parking > avg_price_nocar_parking

Alternate hypthesis: avg_price_car_parking < avg_price_nocar_parking

Significance = 0.05

In [41]:
# with car parking
df_car_parking = df[df['Car']>0]

# without car parking
df_nocar_parking = df[df['Car']==0]


In [42]:
with_car_mean,with_car_std,with_car_sample_size = get_metrics(df_car_parking['Price'])
without_car_mean,withoout_car_std,without_car_sample_size = get_metrics(df_nocar_parking['Price'])


In [43]:
t_value = (with_car_mean - without_car_mean) / mth.sqrt((with_car_std**2 / with_car_sample_size) + (withoout_car_std**2 / without_car_sample_size))
t_value

np.float64(-0.2722470302597254)

In [44]:
t_stat, p_value = stats.ttest_ind(df_car_parking['Price'], df_nocar_parking['Price'], equal_var=False, alternative='greater')

In [45]:
t_stat, p_value

(np.float64(-0.27224703025972535), np.float64(0.60726250073007))

As p-value 0.607 > 0.05, we fail to reject the Null Hypothesis.

We can conclude Developers should prioritize parking as it adds measurable value.

Q8. Two-Way ANOVA (Location & Property Type)
Investigate whether property prices are influenced by:

- Suburb

- Type of property

- Interaction between suburb and property type

Use Two-Way ANOVA and explain which factors significantly affect price

null hypothesis

H0: There is no influence on the property price between different property types.

H1 : There is no influence on the property price between different Suburb.

H2 : There is no interaction between suburb and property type.

Alternate hypothesis:

HA0: There is influence on the property price between different property types.

HA1 : There is influence on the property price between different Suburb.

HA2 : There is some interaction between suburb and property type.






In [46]:
df_annnova = df.groupby(by=['Type','Suburb'])['Price'].agg(['count','mean'])
df_annnova

count          mean
Type Suburb                                 
h    Abbotsford             40  1.216662e+06
     Aberfeldie             32  1.529531e+06
     Airport West           44  8.119091e+05
     Albanvale               5  5.615000e+05
     Albert Park            63  2.032206e+06
...                        ...           ...
u    Williamstown           17  5.116765e+05
     Williamstown North      6  4.064167e+05
     Windsor                34  5.681618e+05
     Wollert                 1  3.900000e+05
     Yarraville             16  4.778715e+05

[690 rows x 2 columns]

In [47]:
df_cleaned_suburb = df['Suburb'].dropna()
df_cleaned_type = df['Type'].dropna()
df_cleaned_price = df['Price'].dropna()

In [48]:
df_cleaned = pd.DataFrame({'Suburb':df_cleaned_suburb, 'Type':df_cleaned_type, 'Price':df_cleaned_price})

In [49]:
df_cleaned.head()

,Suburb,Type,Price
0,Abbotsford,h,1480000.0
1,Abbotsford,h,1035000.0
2,Abbotsford,h,1465000.0
3,Abbotsford,h,850000.0
4,Abbotsford,h,1600000.0


In [50]:
model = ols('Price ~ C(Type) + C(Suburb) + C(Type):C(Suburb)', data=df_cleaned).fit()
anova_table = sm.stats.anova_lm(model, typ=2)

print(anova_table)

/usr/local/lib/python3.12/dist-packages/statsmodels/base/model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 313, but rank is 162
  warnings.warn('covariance of constraints does not have full '


                         sum_sq       df            F         PR(>F)
C(Type)            7.356463e+14      2.0  2291.961008   0.000000e+00
C(Suburb)          3.650621e+15    313.0    72.675929   0.000000e+00
C(Type):C(Suburb)  6.470034e+14    626.0     6.440216  2.505093e-263
Residual           2.068639e+15  12890.0          NaN            NaN


/usr/local/lib/python3.12/dist-packages/statsmodels/base/model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 626, but rank is 375
  warnings.warn('covariance of constraints does not have full '


As the p-value is significantly less than 0.05 for all the three hypothesis, hence rejecting the null hypothesis.

So we can conclude that type of property and suburb affect the price of the property and there is an interaction between suburb and type of property.

Q9. p-Value Interpretation (Decision Making)
A hypothesis test comparing prices across two suburbs results in a p-value of 0.032.
Answer:

• What does this p-value indicate?

• Should the null hypothesis be rejected at α = 0.05?

• How should a business stakeholder interpret this result?

The p-value of 0.032 indicates that if there were actually no difference in property prices between the two suburbs (the null hypothesis), there is only a 3.2% chance that we would see a price difference as large as the one we observed due to random sampling noise.

As the p-value 0.32 is less than 0.05, hence rejecting the null hypothesis.

Because we rejected the null hypothesis, there is a difference in pricing in 2 suburbs

Q10. Industry-Style Hypothesis Validation (Policy Decision)
A housing policy group believes that properties with more than 2 bathrooms
command a premium price.
Design and execute a statistical test to validate this claim:

• Identify the correct test

• State hypotheses

• Report p-value

• Give a clear recommendation to policymakers

Null hypothesis : There is a difference in the price of the property on the bathroom count.

Alternate hypothesis : There is no difference in the price of the property on the bathroom count.

Significance = 0.05

In [51]:
df_2bathrooms = df[df['Bathroom']>2]
df_bathrooms = df[df['Bathroom']<=2]

In [52]:
df_2bathrooms['Price'].describe()

,Price
count,1.060000e+03
mean,1.882824e+06
std,9.766571e+05
min,4.500000e+05
25%,1.170000e+06
50%,1.661500e+06
75%,2.415000e+06
max,8.000000e+06


In [53]:
df_bathrooms['Price'].describe()

,Price
count,1.252000e+04
mean,1.007348e+06
std,5.502773e+05
min,8.500000e+04
25%,6.300000e+05
50%,8.700000e+05
75%,1.266250e+06
max,9.000000e+06


In [54]:
t_stat, p_value = stats.ttest_ind(df_2bathrooms['Price'], df_bathrooms['Price'], equal_var=False, alternative='greater')

In [55]:
t_stat, p_value

(np.float64(28.800231483982945), np.float64(3.69021417834531e-137))

We performed an Independent Two-Sample T-test

We can conclude p-value ($3.69 \times 10^{-137}$) is significantly less than the alpha which 0.05, hence rejecting the null hypothesis.

There is no premium price to pay for the property which has more than 2 bathrooms.